cnn_framework/layers.py

```diff
@@ -12,7 +12,9 @@
 class MoleculeConv(Layer):
     def __init__(self, units, inner_dim, depth=2, init_output='uniform',
                  activation_output='softmax', init_inner='identity',
-                 activation_inner='linear', scale_output=0.01, padding=False, **kwargs):
+                 activation_inner='linear', scale_output=0.01, padding=False, 
+                 dropout_rate_outer=0.0, dropout_rate_inner=0.0,
+                 padding_final_size=None, **kwargs):
```

Comment:
The dropout mask has to be large enough to handle the largest molecule allowed in the model.  So one dimension of the mask has to be the same size as `padding_final_size`.  This is the reason why `padding_final_size` is needed in the definition of `MoleculeConv`.  If `padding_final_size=None` and dropout is used, it will crash.